src/sentry/integrations/custom_scm/repository.py

```diff
@@ -13,8 +16,30 @@ def repository_external_slug(self, repo):
         return repo.name
 
     def dispatch(self, request, organization, **kwargs):
-        # TODO(meredith): Add functionality to actually add a repo
-        # to the manual integration.
-        #   * update provider
-        #   * update integration_id
-        return Response(status=200)
+        """
+        Adding a repository to the Custom SCM integration is
+        just two steps:
+           1. Change the provider from `null` to 'integrations:custom_scm'
+           2. Add the integration_id that is passed from the request
+        """
+        repo_id = request.data.get("identifier")
+        integration_id = request.data.get("installation")
+
+        try:
+            # double check the repository_id passed is not
+            # for an already 'claimed' repository
+            repo = Repository.objects.get(
+                organization_id=organization.id,
+                id=repo_id,
+                integration_id__isnull=True,
+                provider__isnull=True,
+            )
+            integration = Integration.objects.get(organizations=organization, id=integration_id)
+        except (Repository.DoesNotExist, Integration.DoesNotExist):
+            raise Http404
+
+        repo.provider = "integrations:custom_scm"
```

Comment:
@MeredithAnya can we make `"integrations:custom_scm"` a constant?

Reply:
@NisanthanNanthakumar turns out `self.id` of the can we used since it gets set in the setup

https://github.com/getsentry/sentry/blob/50cb7769186bfc9e610c80b47c70e1ba262c92a2/src/sentry/integrations/custom_scm/integration.py#L137-L140